src/estimagic/estimate_ml.py

```diff
@@ -179,6 +185,10 @@ def estimate_ml(
     bounds = pre_process_bounds(bounds)
     jacobian_numdiff_options = pre_process_numdiff_options(jacobian_numdiff_options)
     hessian_numdiff_options = pre_process_numdiff_options(hessian_numdiff_options)
+    # TODO: Overwrite constraints once we deprecate dictionary constraints. We currently
+    # pass constraints unprocessed to the optimizer, so that we do not throw warnings
+    # if the new constraints are used.
+    dict_constraints = pre_process_constraints(constraints)
```

Comment:
I would move this function into `deprecations.py` such that we get an error once we delete `deprecations.py`.